File: second_year/projects/G-OOP-400-COT-4-1-bsraytracer-13/src/ray.cpp
/*
** EPITECH PROJECT, 2026
** snd_year_projects
** File description:
** ray
*/

#include "../include/ray.hpp"
using namespace RayTracer;

Ray::Ray(const Point3D& givenOrigin, const Vector3D& givenDirection)
    : origin(givenOrigin), direction(givenDirection)
{}

File: second_year/projects/G-OOP-400-COT-4-1-bsraytracer-13/rust/src/ray.rs
//
// EPITECH PROJECT, 2026
// snd_year_projects
// File description:
// ray
//

use crate::Vector3D;
use crate::Point3D;

#[derive(Debug, Clone, Copy)]
pub struct Ray {
    pub origin: Point3D,
    pub direction: Vector3D,
}

impl Ray {
    pub fn new(point: Point3D, vector: Vector3D) -> Ray {
        return Ray { origin: point, direction: vector };
    }
}
